core/com.b2international.snowowl.identity/src/com/b2international/snowowl/identity/InternalIdentityProvider.java

```diff
@@ -13,14 +13,18 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.b2international.snowowl.authentication;
-
-import com.b2international.snowowl.core.setup.DefaultBootstrapFragment;
-import com.b2international.snowowl.core.setup.ModuleConfig;
+package com.b2international.snowowl.identity;
 
 /**
- * @since 3.4
+ * @since 5.11
  */
-@ModuleConfig(fieldName = "authentication", type = AuthenticationConfiguration.class)
-public class AuthenticationBootstrap extends DefaultBootstrapFragment {
-}
\ No newline at end of file
+public interface InternalIdentityProvider extends IdentityProvider {
```

Comment:
Instead of nesting the interfaces: `IdentityProvider` &larr; `InternalIdentityProvider` &larr; `XYZIdentityProviderImpl`
This interface could represent mutability: `IdentityProvider` &larr; `XYZIdentityProviderImpl` &rarr; `IdentityWriter`

Reply:
My original thought was to name it `IdentityManager`, but I don't like `*Manager` class names that much. 😄 I like `IdentityWriter`, though, I'll use that. Thanks!